src/PlaneWaveBasis.jl

```diff
@@ -53,15 +54,19 @@ struct PlaneWaveBasis{T <: Real}
     # the basis set is defined by {e_{G}, 1/2|k+G|^2 ≤ Ecut}
     Ecut::T
 
-    # irreducible kpoints
+    # irreducible kpoints. In the case of collinear spin,
+    # this lists all the spin up, then all the spin down
     kpoints::Vector{Kpoint{T}}
-    # BZ integration weights, summing up to 1
-    # kweights[ik] = length(ksymops[ik]) / sum(length(ksymops[ik]) for ik=1:Nk)
+    # BZ integration weights, summing up to model.n_spin_components
     kweights::Vector{T}
     # ksymops[ik] is a list of symmetry operations (S,τ)
     # mapping to points in the reducible BZ
     ksymops::Vector{Vector{SymOp}}
 
+    comm_k::MPI.Comm  # communicator for the kpoints distribution
+    krange_thisproc::Vector{Int}  # indices of kpoints treated explicitly by this
+                                  # processor in the global kcoords array
```

Comment:
thisproc? Not so sure ... how about `krange_local` That's typically used in the MPI jargon.

Also I would not access this field directly but go via a function. That will make it easier if we generalise to other computational backends later (but we can also refactor than of course).

Reply:
if you look at pwbasis and see `krange_local` you might not think it's a parallelization thing. This makes it more explicit. Generalization let's tackle this when we get to it

Reply:
yeah ok. Still not happy about the name, but I also have no better.